src/main/java/adeuxpas/back/service/AdServiceImpl.java

```diff
@@ -126,4 +223,54 @@ public List<AdPostResponseDTO> findAdsByPublisherId(Long publisherId) {
             throw new EntityNotFoundException();
         }
     }
+
+    // ==================== parameter extraction and formatting helper methods =======================
+    private boolean shouldReturnAllAds(List<String> priceRangesFilter, List<String> citiesAndPostalCodesFilter,
+                                       List<String> articleStatesFilter, String categoryFilter) {
+        return priceRangesFilter.isEmpty() && citiesAndPostalCodesFilter.isEmpty() &&
+                articleStatesFilter.isEmpty() && categoryFilter.equals("Catégorie");
+    }
+    private void extractAndAssignCategoryFilterCriteria(String categoryFilter) {
+        if (categoryFilter.contains("▸")) {
+            category = categoryFilter.substring(0, categoryFilter.indexOf(" ▸"));
+            if (categoryFilter.indexOf("▸") != categoryFilter.lastIndexOf("▸")) {
+                subcategory = categoryFilter.substring(categoryFilter.indexOf("▸ ") + 1, categoryFilter.lastIndexOf(" ▸")).trim();
+                gender = categoryFilter.substring(categoryFilter.lastIndexOf("▸ ") + 1).trim();
+            } else
```

Comment:
du coup je me pose la question : 
pour les champs relatifs à la catégorisation dans l'entité Ad, je n'ai mis que la subCategory comme devant être obligatoirement renseignée parce

1.    Le genre ne doit être renseigné if subCat = Hauts || Bas
2.    Une sous-categorie ne peut appartenir qu'à une seule catégorie

Mais vu que le filtrage peut se faire par la categorie seulement, ne faudrait il pas ajouter @NotBlank sur le champs category également ?

Reply:
Comme tu l'as dit, 'Une sous-categorie ne peut appartenir qu'à une seule catégorie'.
En découle d'ici que la catégorie sera connue si on a la souscatégorie.

Donc, soit tu crée une logique dans le back pour en déduire la Category à partir de la Subcategory, soit tu t'attends à recevoir la Category depuis le formulaire et tu enforce cette attente par une validation front (champ obligatoire dans le formulaire) et back (champ not null etc dans le corps de la requête, ou, dans notre cas, dans le DTO).

Le principal c'est que l'article ait obligatoirement une catégorie à la fin .

Et ça m'a fait penser si on ne devrait pas ajouter finalement des validations dans la base de donnée aussi, pour tous les champs que l'on désigne comme obligatoires

C'est une bonne question, il faut en reparler lors du meet